harvest/api/kraken.py

```diff
@@ -0,0 +1,292 @@
+        'CRV': 'CRV',      'ZRX': 'ZRX',       'NANO': 'NANO',
+        'ANKR': 'ANKR',    'SAND': 'SAND',     'REN': 'REN',
+        'KAVA': 'KAVA',    'MINA': 'MINA',     '1INCH': '1INCH',
+        'GHST': 'GHST',    'ANT': 'ANT',       'REP': 'XREP',
+        'REPV2': 'XREPV2', 'BADGER': 'BADGER', 'BAL': 'BAL',
+        'BAND': 'BAND',    'CTSI': 'CTSI',     'CQT': 'CQT',
+        'EWT': 'EWT',      'MLN': 'XMLN',      'ETH2': 'ETH2',
+        'GNO': 'GNO',      'INJ': 'INJ',       'KAR': 'KAR',
+        'KEEP': 'KEEP',    'KNC': 'KNC',       'LSK': 'LSK',
+        'LTP': 'LTP',      'LRC': 'LRC',       'MIR': 'MIR',
+        'OCEAN': 'OCEAN',  'PAXG': 'PAXG',     'RARI': 'RARI',
+        'REN': 'REN',      'XRP': 'XXRP',      'SRM': 'SRM',
+        'STORJ': 'STORJ',  'TBTC': 'TBTC',     'OGN': 'OGN',
+        'OXT': 'OXT'
+    }
+
+
+    def __init__(self, path: str=None):
+        super().__init__(path)
+        self.api = krakenex.API(self.config['api_key'], self.config['secret_key'])
+
+    def no_secret(self, path: str) -> bool:
+        return self.create_secret(path)
+
+    def setup(self, watch: List[str], interval: str, trader=None, trader_main=None):
+        self.watch_crypto = []
+        if is_crypto(s):
+            self.watch_crypto.append(s)
+        else:
+            warning("Kraken does not support stocks!")
+
+        self.option_cache = {}
+        super().setup(watch, interval, interval, trader, trader_main)
+
+    def exit(self):
+        self.option_cache = {}
+
+    def main(self):
+        df_dict = {}
+        df_dict.update(self.fetch_latest_crypto_price())
+
+        self.trader_main(df_dict)
+
+    @API._exception_handler
+    def fetch_latest_crypto_price(self):
+        dfs = {}
+        for symbol in self.watch_cryptos:
+            dfs[symbol] = self.fetch_price_history(symbol, self.interval, now() - dt.timedelta(days=7), now()).iloc[[0]]
+        return dfs
+
+    # -------------- Streamer methods -------------- #
+
+    @API._exception_handler
+    def fetch_price_history( self,
+        symbol: str,
+        interval: str,
+        start: dt.datetime = None,
+        end: dt.datetime = None,
+       ):
+
+        debug(f"Fetching {symbol} {interval} price history")
+
+        if start is None:
+            start = now() - dt.timedelta(days=365 * 5)
+        if end is None:
+            end = now()
+
+        if start >= end:
+            return pd.DataFrame()
+
+        if interval not in self.interval_list:
+            raise Exception(f"Interval {interval} not in interval list. Possible options are: {self.interval_list}")
+        val, unit = expand_interval(interval)
+        df = self.get_data_from_kraken(symbol, val, unit, start, end)
+
+        return df
+
+    @API._exception_handler
+    def fetch_chain_info(self, symbol: str):
+        return {
+            "id": "n/a",
+            "exp_dates": [str_to_date(s) for s in self.watch_ticker[symbol].options],
+            "multiplier": 100
+        }
```

Comment:
What is `fetch_chain_info` supposed to do?

Reply:
This function is for options, so no need to implement for cryptos